src/components/SearchBar.tsx

```diff
@@ -18,11 +18,18 @@ export default function SearchBar() {
 
   useDebounce(() => setSearchText(value), 200, [value])
 
-  document.body.addEventListener("keydown", (event) => {
+  const searchOverride = (event: KeyboardEvent) => {
     if (event.ctrlKey && event.key === "f") {
       event.preventDefault()
       document.getElementById(id)?.focus()
     }
+  }
+
+  useEffect(() => {
+    document.body.addEventListener("keydown", searchOverride)
+    return () => {
+      document.body.removeEventListener("keydown", searchOverride)
+    }
   })
```

Comment:
The event listener gets set every render, because the effect has no dependency array. 

Could you add a dependency array, with the `id` variable in the array and move the definition of `searchOverride within the effect? This way, the event handler is only re-added when the id changes (which it rarely does)

Like this:
```typescript
useEffect(() => {
  const searchOverride = (event: KeyboardEvent) => {
    if (event.ctrlKey && event.key === "f") {
      event.preventDefault()
      document.getElementById(id)?.focus()
    }
  }
  
  document.body.addEventListener("keydown", searchOverride)

  return () => {
    document.body.removeEventListener("keydown", searchOverride)
  }
}, [id])
```